Factor out eager val from eval_llama_lib

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #3757
* __->__ #3756

Would like to re-use EagerEvalWrapper and eval function for quantization calibration.

Differential Revision: [D57881028](https://our.internmc.facebook.com/intern/diff/D57881028/)

cccclai committed:
Factor out eager val from eval_llama_lib

Would like to re-use EagerEvalWrapper and eval function for quantization calibration.

Differential Revision: [D57881028](https://our.internmc.facebook.com/intern/diff/D57881028/)

[ghstack-poisoned]

cccclai committed:
Update on "Factor out eager val from eval_llama_lib"


Would like to re-use EagerEvalWrapper and eval function for quantization calibration.

Differential Revision: [D57881028](https://our.internmc.facebook.com/intern/diff/D57881028/)

[ghstack-poisoned]

cccclai committed:
Update on "Factor out eager val from eval_llama_lib"


Would like to re-use EagerEvalWrapper and eval function for quantization calibration.

Differential Revision: [D57881028](https://our.internmc.facebook.com/intern/diff/D57881028/)

[ghstack-poisoned]

cccclai committed:
Update on "Factor out eager val from eval_llama_lib"


Would like to re-use EagerEvalWrapper and eval function for quantization calibration.

Differential Revision: [D57881028](https://our.internmc.facebook.com/intern/diff/D57881028/)

[ghstack-poisoned]

Jack-Khuu commented on examples/models/llama2/builder.py:
```diff
@@ -15,6 +15,13 @@
+
+try:
+    from ...portable.utils import export_to_edge, save_pte_program
+except ImportError:
```
let's put this as a task to fix separately. I've seem this issue in other parts of the repo too

cccclai replied:
Yeah makes sense. Let me update

Jack-Khuu reviewed (approved):
Thanks for the refactor, makes things easier to reference

cccclai committed:
Update on "Factor out eager val from eval_llama_lib"


Would like to re-use EagerEvalWrapper and eval function for quantization calibration.

Differential Revision: [D57881028](https://our.internmc.facebook.com/intern/diff/D57881028/)

[ghstack-poisoned]
